blueocean-dashboard/src/main/js/components/RunDetails.jsx

```diff
@@ -137,10 +138,13 @@ class RunDetails extends Component {
     afterClose = () => {
         const { router, params } = this.context;
         if (this.opener) {
-            router.goBack();
+            // step back in the history to the item that's prior to initial load of RunDetails
+            const offsetToInitialRoute = this.initialHistoryLength - history.length;
+            router.go(offsetToInitialRoute - 1);
```

Comment:
the problem is the same as tracking the opener. you could navigate around, close the modal because you navigate to the editor for example, then go back to the modal and all of a sudden the `initialHistoryLength` is recalculated, since it remounted...